src/base/misc/fatfs.c

```diff
@@ -1572,7 +1574,14 @@ void build_boot_blk(fatfs_t *f, unsigned char *b)
     case OLDMSD_D:
       txfr->ofs = 0x0000;
       txfr->seg = 0x0070;
-      txfr->ax  = d_o >> 16;
+      /*
+       * MS-DOS 3.10 needs the offset of MSDOS.SYS / 16 passed in AX and other
+       * versions don't seem to mind. See the issue discussion at
+       * https://github.com/stsp/dosemu2/issues/278. Note this relies upon the
+       * ordering of the directory, as was a requirement with earlier DOSes.
+       */
+      clstrsize = f->cluster_secs * SECTOR_SIZE;
+      txfr->ax  = ((f->obj[1].size + clstrsize - 1) & ~clstrsize) / 16;
```

Comment:
It's better if you fix it, that way it'll look how you want it and we'll save the round trips.